### DISCUSSION TITLE
Dock Control Support

### DISCUSSION THREAD

**Discussion by chasebolt** [humbertogontijo/homeassistant-roborock#438] [General]
Is it possible to use the dock control, such as Empty / Wash / Stop?

**Comment by arne182** [comment]
Dock empty does have a service command that work. But I have not found the one for wash.

**Comment by arne182** [comment]
vacuum.send_command
app_start_wash
app_stop_wash
app_start_collect_dust
Are working but 
start_wash_then_charge
Does not work

**Reply by chasebolt** [reply]
how did you find these commands? are they documented somewhere?

**Reply by arne182** [reply]
https://github.com/humbertogontijo/python-roborock/blob/main/roborock/roborock_typing.py

**Comment by arne182** [comment]
I have a S8 pro Ultra
Hast anyone gotten this command to work?
start_wash_then_charge

**Reply by chasebolt** [reply]
i have Q Revo and this command does not work for me. what is it supposed to do?

**Comment by arne182** [comment]
Looks like app_start_wash does auto dock the S8 after it is finished

**Comment by arne182** [comment]
Does anyone know how the APP_SET_DRYER_SETTING works?

**Reply by rossi75** [reply]
You have the Dryer add-on? Hence there is a sensor which is triggered by the S8s firmware via one of the IR crash sensors. Well, may I contact you for a mission for the S7 pro ultra?
